connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2739,7 +2739,7 @@ class Dataset[T] private[sql] (
     sparkSession.analyze(plan, proto.AnalyzePlanRequest.AnalyzeCase.SCHEMA)
   }
 
-  def collectResult(): SparkResult[T] = sparkSession.execute(plan, encoder)
+  private[sql] def collectResult(): SparkResult[T] = sparkSession.execute(plan, encoder)
```

Comment:
As mentioned in the other PR. Please do not change this.

Reply:
`checkDatasetApiCompatibility('compatibility API tests: Dataset' in CompatibilitySuite)` will check find out the public Dataset api in `connect/client` module  instead of in `sql` module. `collectResult ` is such a case. There was no problem before because there was bad case in the test (classloader loaded the wrong class)

Reply:
> I guess it is because this is not an existing API?

Yes

Reply:
I think If we accept similar changes (add some new public apis to `connect/client` module), we may need to maintain a whitelist(in checkDatasetApiCompatibility) or do not check this again



checkDatasetApiCompatibility

Reply:
This is one of the only places where we will be deviating from the original API. I do intent to create the same class for the original API, but that is for later.

Reply:
without this change, this check will print error message as follows:

```
ERROR: Dataset apis incompatible with the sql module include: 
org.apache.spark.sql.Dataset.collectResult
connect-client-jvm module mima check failed.
Exceptions to binary compatibility can be added in tools/CheckConnectJvmClientCompatibility.scala
```

Reply:
It comes to the question of do we pursue this check to verify that Scala client is a subset of the original API or intersect with it. Per my understanding, current implementation is to check the subset.

Reply:
> changed to check intersection

What does this mean? Is the intersection compatible? It seems that `checkMiMaCompatibility` has already done this?

Reply:
I will leave @hvanhovell to comment and see the right way to address this. 

To be clear I was thinking based on this
```
    // For now we simply check the new methods is a subset of the old methods.
    clientMethods.diff(sqlMethods)
  }
```

I thought the comment says that we only check if new methods are a subset of the existing methods otherwise it will fail. And you need to change the method otherwise it becomes not a subset.

Reply:
Do you mean we should change this to 

```
    clientMethods.filterNot(m => sqlMethods.contains(m))
```
?

Reply:
```scala
    // For now we simply check the new methods is a subset of the old methods.
    clientMethods.diff(sqlMethods)
```

`the new methods is a subset of the old method`  -> `clientMethods.diff(sqlMethods)` should be empty?

For example:

```
scala> val clientMethods = Seq(1,2,3)
clientMethods: Seq[Int] = List(1, 2, 3)

scala> val sqlMethods = Seq(1,2,3,4,5)
sqlMethods: Seq[Int] = List(1, 2, 3, 4, 5)

scala> clientMethods.diff(sqlMethods)
res0: Seq[Int] = List()
```

`clientMethods` is a subset of `sqlMethods`.


```
scala> val clientMethods = Seq(0,1,2,3)
clientMethods: Seq[Int] = List(0, 1, 2, 3)

scala> val sqlMethods = Seq(1,2,3,4,5)
sqlMethods: Seq[Int] = List(1, 2, 3, 4, 5)

scala> clientMethods.diff(sqlMethods)
res0: Seq[Int] = List(0)
```

`clientMethods` is not a subset of `sqlMethods`.

Reply:
The original test code as follows. I think it is consistent with the current intention, am I right?

```
// For now we simply check the new methods is a subset of the old methods.
    newMethods
      .map(m => m.toString)
      .foreach(method => {
        assert(oldMethods.map(m => m.toString).contains(method))
      })
```

Reply:
Please feel free to ignore me and work @hvanhovell on this case. :)

Reply:
@LuciferYang can we make an exception for the SparkResult class. Basically filter out the DataSet.collectResult method. That is all.

Reply:
let me add a white list to `checkDatasetApiCompatibility `